flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosResourceManager.java

```diff
@@ -391,13 +391,7 @@ private void recoverWorkers(final List<MesosWorkerStore.Worker> tasksFromPreviou
 
 	@Override
 	public CompletableFuture<Void> postStop() {
-		final CompletableFuture<Void> supportActorsStopFuture = stopSupportingActorsAsync();
-
-		final CompletableFuture<Void> terminationFuture = super.postStop();
-
-		return supportActorsStopFuture.thenCombine(
-			terminationFuture,
-			(Void voidA, Void voidB) -> null);
+		return super.postStop().thenCompose((ignored) -> stopSupportingActorsAsync());
```

Comment:
I think it should be the other way around. First stop sub class resources and then parent class resources.

Reply:
m(